frontend/public/components/utils/resource-link.tsx

```diff
@@ -68,49 +72,118 @@ export const resourcePath = (kind: K8sResourceKindReference, name?: string, name
 export const resourceObjPath = (obj: K8sResourceKind, kind: K8sResourceKindReference) =>
   resourcePath(kind, _.get(obj, 'metadata.name'), _.get(obj, 'metadata.namespace'));
 
-export const ResourceLink = connectToModel(
-  ({
-    className,
-    displayName,
-    inline = false,
-    kind,
-    linkTo = true,
-    name,
-    namespace,
-    hideIcon,
-    title,
-    children,
-  }) => {
-    if (!kind) {
-      return null;
+const resourceLinkStateToProps = (state: RootState): resourceLinkStateProps => {
+  return {
+    activeNamespace: state.UI.get('activeNamespace'),
+    receivedCSVs: state.k8s.getIn(['CSVS', 'csvs']),
+  };
+};
+
+export const compareKinds = (csv, owner) => {
+  return (
+    csv.spec.customresourcedefinitions.owned.some((api) => api.kind === owner.kind) &&
+    csv.spec.customresourcedefinitions.owned.some((api) => owner.apiVersion.includes(api.version))
+  );
+};
+
+const buildManagerLink = (csvName, group, kind, namespace, name, owner, version) => {
+  const reference = `${group}~${version}~${kind}`;
+
+  const link = `/k8s/ns/${namespace}/${ClusterServiceVersionModel.plural}/${csvName}/${reference}/${name}`;
+
+  return link;
+};
+
+const matchOwnerAndCSV = (activeNamespace, owner, receivedCSVs = []) => {
+  const namespacedCSVs = receivedCSVs.filter((csv) => csv.metadata.namespace === activeNamespace);
+  const csvs = namespacedCSVs.filter((csv) => compareKinds(csv, owner));
+  if (csvs.length === 0) {
+    return null;
+  }
+  return csvs[0];
+};
+
+export const namespacedResourceLink = ({
+  activeNamespace,
+  className,
+  displayName,
+  inline = false,
+  kind,
+  linkTo = true,
+  name,
+  namespace,
+  hideIcon,
+  title,
+  children,
+  receivedCSVs,
+  owner = null,
+}: {
+  activeNamespace?: string;
+  className: string;
+  displayName: string;
+  inline: boolean;
+  kind: any;
+  linkTo: boolean;
+  name: string;
+  namespace: string;
+  hideIcon: boolean;
+  title: string;
+  children: any;
+  receivedCSVs?: any;
+  owner?: any;
+}) => {
+  if (!kind) {
+    return null;
+  }
+  let path = resourcePath(kind, name, namespace);
+  let value = displayName ? displayName : name;
+  let iconKind = kind;
+
+  if (owner !== null) {
+    const managingCSV = matchOwnerAndCSV(activeNamespace, owner, receivedCSVs);
+    if (managingCSV !== null) {
+      path = buildManagerLink(
+        managingCSV.metadata.name,
+        owner.apiVersion.split('/')[0],
+        owner.kind,
+        activeNamespace,
+        owner.name,
+        owner,
+        owner.apiVersion.split('/')[1],
+      );
+      value = owner.name;
+      iconKind = referenceForOwnerRef(owner);
```

Comment:
`value` and `iconKind` should not change. We always want to show the value for the resource, not the owner. Only the link URL should change.